pkg/kv/kvserver/store_split.go

```diff
@@ -130,21 +130,14 @@ func splitPreApply(
 		return
 	}
 
-	// Update the raft HardState with the new Commit value now that the
-	// replica is initialized (combining it with existing or default
-	// Term and Vote). This is the common case.
+	// The RHS replica exists and is uninitialized. We are initializing it here.
+	// Update the raft HardState with the new Commit index (taken from the applied
+	// state in the write batch), and use existing or default Term and Vote. This
+	// is the common case.
 	rsl := stateloader.Make(split.RightDesc.RangeID)
 	if err := rsl.SynthesizeRaftState(ctx, readWriter); err != nil {
 		log.Fatalf(ctx, "%v", err)
 	}
-	// Write the RaftReplicaID for the RHS to maintain the invariant that any
-	// replica (uninitialized or initialized), with persistent state, has a
-	// RaftReplicaID. NB: this invariant will not be universally true until we
-	// introduce node startup code that will write this value for existing
```

Comment:
We have already passed this stage: all replicas have a `RaftReplicaID` in storage.